azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -268,18 +269,18 @@ private boolean handleNonReadyStatus() {
     if (Status.isStatusFinished(nodeStatus)) {
       quickFinish = true;
     } else if (nodeStatus == Status.DISABLED) {
-      changeStatus(Status.SKIPPED, time);
+      nodeStatus = changeStatus(Status.SKIPPED, time);
       quickFinish = true;
     } else if (this.isKilled()) {
-      changeStatus(Status.KILLED, time);
+      nodeStatus = changeStatus(Status.KILLED, time);
       quickFinish = true;
     }
 
     if (quickFinish) {
       node.setStartTime(time);
-      fireEvent(Event.create(this, Type.JOB_STARTED, null, false));
```

Comment:
Why this change? The old code sets shouldUpdate to false. You removed it. Could you explain your design?